ios/Ciphers/AesCipher.swift

```diff
@@ -10,124 +10,248 @@ import Foundation
-            
-            // Everything ok, close streams
-            input.close()
-            output.close();
-            
-            callback(nil,status)
-        
-        
+
+    let cryptStream = StreamCryptor(
+      operation: StreamCryptor.Operation.encrypt,
+      algorithm: algorithm,
+      mode: StreamCryptor.Mode.CTR,
+      padding: StreamCryptor.Padding.NoPadding,
+      key: key,
+      iv: iv
+    )
+
+    // Prepare buffers
+    var inputBuffer = [UInt8](repeating: 0, count: bufferSize)
+    var outputBuffer = [UInt8](repeating: 0, count: bufferSize)
+
+    // Open streams
+    input.open()
+    output.open()
+
+    var encryptedBytes: Int = 0
+    while input.hasBytesAvailable {
+      // Read the bytes
+      let bytesRead = input.read(&inputBuffer, maxLength: inputBuffer.count)
+      let status = cryptStream.update(
+        bufferIn: inputBuffer, byteCountIn: bytesRead, bufferOut: &outputBuffer,
+        byteCapacityOut: outputBuffer.count, byteCountOut: &encryptedBytes)
+
+      if status != Status.success {
+        // Handle this state, close streams and
+        // notify via callback maybe?
+      }
+      // Make sure status is Ok
+      if encryptedBytes > 0 {
+        let bytesOut = output.write(outputBuffer, maxLength: encryptedBytes)
+        assert(bytesOut == Int(encryptedBytes))
+      }
+
     }
-    
-    
-    // Decrypts an input stream to an output stream given a key and an IV
-    // using AES256 CTR mode with No padding
-    
-    public func decrypt(input: InputStream, output:OutputStream, key: [UInt8], iv: [UInt8], callback: (_:RnCryptoError?, _: Status?) -> Void ) -> Void {
-   
-        if !self.isValidIv(iv: iv) {
-            return callback(RnCryptoError.badIv, nil)
+
+    // Final check
+    let status = cryptStream.final(
+      bufferOut: &outputBuffer, byteCapacityOut: outputBuffer.count,
+      byteCountOut: &encryptedBytes)
+
+    // Everything ok, close streams
+    input.close()
+    output.close()
+
+    callback(nil, status)
+
+  }
+
+  // Encrypts an input stream to multiple output streams in chunks
+  public func encryptToChunks(
+    input: InputStream,
+    outputs: [OutputStream],
+    key: [UInt8],
+    iv: [UInt8],
+    chunkSize: Int,
+    callback: (_ error: RnCryptoError?, _ status: Status?) -> Void
+  ) {
+    let algorithm = StreamCryptor.Algorithm.aes
+    let bufferSize: Int = 1024 * 8
+    guard isValidIv(iv: iv) else {
+      return callback(.badIv, nil)
+    }
+
+    let cryptStream = StreamCryptor(
+      operation: .encrypt,
+      algorithm: algorithm,
+      mode: .CTR,
+      padding: .NoPadding,
+      key: key,
+      iv: iv
+    )
+
+    var inputBuffer = [UInt8](repeating: 0, count: bufferSize)
+    var outputBuffer = [UInt8](repeating: 0, count: bufferSize)
+    input.open()
+    var encryptedBytes: Int = 0
+    var chunkIndex = 0
+    var bytesWrittenInChunk = 0
+    var totalBytesProcessed: Int64 = 0
+
```

Comment:
I would add a guard also for validating the `chunkSize` value to ensure is not < than the file size, for instance.

Reply:
I'm sorry, I meant bigger @CandelR